apps/system/test/unit/task_manager_test.js

```diff
@@ -1,1377 +1,570 @@
+        );
+        assert.equal(tm.currentCard.element.style.transform,
+                     `translate(${expectedLeft}px, 0px)`);
       });
 
-      MockService.mockQueryWith('AppWindowManager.getActiveWindow', apps.sms);
-      _filterName = 'browser-only';
-      stub = this.sinon.stub(taskManager, 'filter', function(filterName) {
-          assert.equal(filterName, _filterName);
-          taskManager.stack = [];
-        });
-    });
+      test('centering at different stack position after closing',
+      function(done) {
+        // close and re-open the task manager at a different stack position
+        tm.hide(apps.game).then(() => {
+          // open again at position 0.
+          MockStackManager.mCurrent = 0;
+          return tm.show();
+        }).then(() => {
+          var currentApp = MockStackManager.mStack[0];
+          var elem = tm.currentCard.element;
 
-    test('filter function is called and empty stack is the result', function() {
-      showTaskManager(this.sinon.clock, _filterName);
-      stub.calledWith([_filterName]);
-      assert.isTrue(cardsView.classList.contains('empty'),
-                    'Should be displaying no recent browser windows');
-    });
+          assert.equal(currentApp, tm.currentCard.app);
 
-    test('but apps should still enterTaskManager', function() {
-      MockStackManager.mStack.forEach((app) => {
-        this.sinon.spy(app, 'enterTaskManager');
+          var expectedLeft =
+            getExpectedCardPlacement(elem, MockStackManager.mCurrent);
+          assert.equal(tm.currentCard.element.style.transform,
+                       `translate(${expectedLeft}px, 0px)`);
+        }).then(done, done);
+        clock.tick(1000);
       });
-
-      showTaskManager(this.sinon.clock, _filterName);
-
-      MockStackManager.mStack.forEach(function(app) {
-        sinon.assert.calledOnce(app.enterTaskManager);
-      }, this);
     });
 
-  });
 
-  suite('orientation', function() {
-    setup(function() {
-      MockStackManager.mStack = [
-        appConfigs.sms,
-      ].map((config) => {
-        var app = new AppWindow(config);
-        apps[app.name] = app;
-        return app;
-      });
-      MockStackManager.mCurrent = 0;
-      MockService.mockQueryWith('AppWindowManager.getActiveWindow', apps.sms);
-      MockStackManager.mCurrent = 0;
-    });
 
-    test('lock orientation when showing', function() {
-      var orientation = MockService.mockQueryWith('defaultOrientation');
-      this.sinon.stub(screen, 'mozLockOrientation');
-      showTaskManager(this.sinon.clock);
-      assert.isTrue(screen.mozLockOrientation.calledWith(orientation));
+    test('kill cards and update', function() {
+      MockStackManager.mStack.length = 2;
+      tm.handleEvent(new CustomEvent('appterminated'));
+      clock.tick(1000);
+      assert.equal(
+        document.querySelectorAll('.card').length,
+        MockStackManager.mStack.length
+      );
     });
```

Comment:
I'd like a second test here dispatching a click event on the card's close button to close instead to trigger the card removal